### ISSUE TITLE
remove core-js from dependencies

### ISSUE THREAD

**Issue by jmarucha** [joepuzzo/informed#88]
The core-js polyfills from babel-transform-plugin are unnecessary for many targets (e.g. someone is not compiling for IE11). This is kind of against idea of being lightweight, as the dependency makes the project 150kB bigger just by adding core-js which is sometimes unnecessary.
One solution is to rely on ES6 and ES7 features and recommend using babel-polyfill, and another is to publish untranspiled code and leave it for user to transpile as necessary.

**Comment by joepuzzo** [comment]
Hmm so according do docs https://babeljs.io/docs/en/babel-plugin-transform-runtime i need that polyfill at runtime for the code to work. The issue is if someone never used async in there code, and all of a sudden pulled in informed (uses async), there code would be broken. Because of this i transpile it with https://babeljs.io/docs/en/babel-plugin-transform-runtime. Any suggestions on how i can transpile without this dep and still have others code work?

**Comment by jmarucha** [comment]
So I took a closer look and noticed you are using webpack to create bundle which is later published as NPM module - this is not a perfect approach, as you convert your library into a huge monolith, and when user uses Webpack to build it's own app, it includes whole code, preventing tree shaking. The better approach (in my opinion) is to publish unbundled, but transpiled code. This will make it possible to include only the parts of Informed CoreJS which are actually included.
Also, publishing both transpiled and untranspiled code is solution to support both ES5 and ES6 users without requiring them to use polyfills if they don't need them. This is described here: https://stackoverflow.com/a/52509974

**Comment by joepuzzo** [comment]
So im not actually bundling the node modules with it. If you look at this line https://github.com/joepuzzo/informed/blob/master/config/webpack/webpack.dist.config.js#L9 in the webpack config, you can see its ignoring node modules for the bundle

**Comment by joepuzzo** [comment]
@joeljeske so i would rather not have multiple versions published. Whatever the best convention is and what most people want is how i would like to bundle this lib. I am however starting to be convinced that i should not bundle the app with the core.js deps and leave it up to the dev to do that.. only reason that sucks is because i cant just tell users to head over to code sandbox and simply add informed as a dep and all will work. 

Basically my goal is to make using informed as easy as`npm install informed` and users can play. No configuration :(

**Comment by joepuzzo** [comment]
@joeljeske so i would rather not have multiple versions published. Whatever the best convention is and what most people want is how i would like to bundle this lib. I am however starting to be convinced that i should not bundle the app with the core.js deps and leave it up to the dev to do that.. only reason that sucks is because i cant just tell users to head over to code sandbox and simply add informed as a dep and all will work.
Basically my goal is to make using informed as easy asnpm install informed and users can play. No configuration :(

**Comment by joeljeske** [comment]
Sorry, I think I was rather confusing. Just to be clear, I am not suggesting multiple packages or versions on NPM, rather 2 sets of code inside the single archive that the bundler can choose based on the package.json (main vs module). 

That being said, I think it would still be safe to remove the core-js dependency and still be able to run on code sandbox as nearly all devs are going to be using evergreen browsers that have the latest ES features. It would just be up to them to provide JS polyfills if using on older IE browsers. 

If you would like, I can try to make a PR.

**Comment by joepuzzo** [comment]
I now dont bundle corejs and also export an esm version of the lib for tree shaking!!
